plugins/modules/pnpm.py

```diff
@@ -0,0 +1,412 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2023 Aritra Sen <aretrosen@proton.me>
+# Copyright (c) 2017 Chris Hoffman <christopher.hoffman@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = """
+---
+module: pnpm
+short_description: Manage node.js packages with pnpm
+description:
+  - Manage node.js packages with the pnpm package manager (https://pnpm.io/)
+author:
+  - "Aritra Sen (@aretrosen)"
+  - "Chris Hoffman (@chrishoffman), creator of NPM Ansible module)"
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  name:
+    description:
+      - The name of a node.js library to install
+      - All packages in package.json are installed if not provided.
+    type: str
+    required: false
+  path:
+    description:
+      - The base path to install the node.js libraries.
+    type: path
+    required: false
+  version:
+    description:
+      - The version of the library to be installed, in semver format.
+    type: str
+    required: false
+  global:
+    description:
+      - Install the node.js library globally
+    required: false
+    default: false
+    type: bool
+  executable:
+    description:
+      - The executable location for pnpm.
+      - The default location it searches for is $PNPM_HOME, fails if not set.
+    type: path
+    required: false
+  ignore_scripts:
+    description:
+      - Use the --ignore-scripts flag when installing.
```

Comment:
```suggestion
      - Use the C(--ignore-scripts) flag when installing.
```
See https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#format-macros-within-module-documentation and sections near it for more information on Ansible markup.